### PR TITLE
[4] GET request scheduled task plugin - Authorization 401

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue # .

### Summary of Changes
fix header for Authorization

### Testing Instructions

- create a GET request  scheduled task
- fill a request URL
- enable Authorization
- fill the key
- run the task

### Actual result BEFORE applying this Pull Request
401

### Expected result AFTER applying this Pull Request
200

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@alikon I think you have to adapt the expected result of the unit tests (file `tests/Unit/Plugin/Task/Requests/Extension/RequestsPluginTest.php` line 242) to your change, see https://ci.joomla.org/joomla/joomla-cms/65072/1/11 .

File: tests/Unit/Plugin/Task/Requests/Extension/RequestsPluginTest.php [hunk 1]
```diff
@@ -239,7 +239,7 @@ public static function isSupported()
         );
         $plugin->standardRoutineHandler($event);
 
-        $this->assertEquals(['basic' => '123'], $transport->headers);
+        $this->assertEquals(['Authorization' => ' basic 123'], $transport->headers);
```
Comment (richard67) [inline tests/Unit/Plugin/Task/Requests/Extension/RequestsPluginTest.php]:
```suggestion
        $this->assertEquals(['Authorization' => 'basic 123'], $transport->headers);
```
Fix failing unit test.

**Review by richard67 — CHANGES_REQUESTED** [review]

File: tests/Unit/Plugin/Task/Requests/Extension/RequestsPluginTest.php (hunk 1 above)
Comment (alikon) [reply tests/Unit/Plugin/Task/Requests/Extension/RequestsPluginTest.php]: oops

**Review by richard67 — APPROVED** [review]

**Comment by richard67** [comment]
Unit tests are passing. Failing system tests in drone are not related to this PR. I will restart later to see if they succeed.

**Comment by richard67** [comment]
@alikon In the testing instructions you wrote:

> create a GET request scheduled task plugin

But I think it just needs to create a task, not write an own plugin, is that right?

**Comment by alikon** [comment]
yes

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 759f44395261a2a985fa183ad435e354eb4904a7<br /><br />I have used a hack for testing with Basic authentication, which is not available in the dropdown for the authentication type. So I have set in database in the task's params the "authType" to "Basic" and the "authKey" to the bease64 encoded combination of user name and password separated by a colon. This worked, i.e. without the PR applied I got a 401 and with the PR applied a 200.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40508">issues.joomla.org/tracker/joomla-cms/40508</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 759f44395261a2a985fa183ad435e354eb4904a7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40508">issues.joomla.org/tracker/joomla-cms/40508</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40508">issues.joomla.org/tracker/joomla-cms/40508</a>.</sub>

**Comment by obuisard** [comment]
Thank you Nicola @alikon for the PR!

**Pull request merged**